app/src/ui/open-pull-request/pull-request-files-changed.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react'
+import { IDiff, ImageDiffType } from '../../models/diff'
+import { Repository } from '../../models/repository'
+import { CommittedFileChange } from '../../models/status'
+import { SeamlessDiffSwitcher } from '../diff/seamless-diff-switcher'
+import { Dispatcher } from '../dispatcher'
+import { openFile } from '../lib/open-file'
+
+interface IPullRequestFilesChangedProps {
+  readonly repository: Repository
+  readonly dispatcher: Dispatcher
+
+  /** The file whose diff should be displayed. */
+  readonly selectedFile: CommittedFileChange | null
+
+  /** The diff that should be rendered */
+  readonly diff: IDiff | null
+
+  /** The type of image diff to display. */
+  readonly imageDiffType: ImageDiffType
+
+  /** Whether we should display side by side diffs. */
+  readonly showSideBySideDiff: boolean
+
+  /** Whether we should hide whitespace in diff. */
+  readonly hideWhitespaceInDiff: boolean
+}
+
+/**
+ * A component for viewing the file diff for a pull request.
+ */
+export class PullRequestFilesChanged extends React.Component<
+  IPullRequestFilesChangedProps,
+  {}
+> {
+  /**
+   * Opens a binary file in an the system-assigned application for
+   * said file type.
+   */
+  private onOpenBinaryFile = (fullPath: string) => {
+    openFile(fullPath, this.props.dispatcher)
+  }
+
+  /** Called when the user changes the hide whitespace in diffs setting. */
+  private onHideWhitespaceInDiffChanged = (hideWhitespaceInDiff: boolean) => {
+    const { selectedFile } = this.props
+    return this.props.dispatcher.onHideWhitespaceInHistoryDiffChanged(
+      hideWhitespaceInDiff,
+      this.props.repository,
+      selectedFile as CommittedFileChange
```

Comment:
Do you need to cast this? `selectedFile` should already be `CommittedFileChange` (or `null`, but that's accepted too).
```suggestion
      selectedFile
```

Reply:
I copied this from somewhere from selected commit and didn't question it (should have 😞). I am guessing at one point null wasn't handled.. I will catch this in another PR.